### DISCUSSION TITLE
Two amateur questions. Samples and book.

### DISCUSSION THREAD

**Discussion by wc-yaa** [VBAndCs/sVB-Small-Visual-Basic#52] [General]
Hello, I have two questions. 

There are many nice samples in the package. Really good. Some of them are games(ball game, cars, etc.) If several famous classic games are also placed in the package, children may easily be motivated to try to learn this BASIC programming language. For example, Tic-tac-toe, Tetris, Snake, Asteroids, Pong, etc. Let children know the versatility of this BASIC language. In the past, they probably only thought that only GameMaker, Scratch or Godot could be used to make games.

Another question is about book for children. There is a nice Python book for children. The book was published many years ago and is quite well-known. The book teaches readers how to make about 10 mini-games. In the process of learning how to make each game, the readers can also learn several programming concepts. The book can be bought in online shops but the HTML version can be read here:
https://inventwithpython.com/invent4thed/

This arrangement is nice. Children can read the book freely. This is also an effective promotion for the book and Python. Right? 
If the author of Small Visual Basic plans to write a new book, this type of book may be a good option. 

The author of Small Visual Basic may be too busy in developing Small Visual Basic. Thus I really hope that some experienced Small Visual Basic users will also consider the above idea. It is a meaningful concept. 

(Actually, the writer wrote another book several years later: https://inventwithpython.com/pygame/)

**Comment by VBAndCs** [comment]
Thanks for your passion for sVB.
I started writing a book series for kids. The first book is for 6 years old. I am using a deferent stratigy based on my feedback from some courses I gave to groups of children. Unfortunately, games ang graphics needs a good understanding of geometry and coordinates, which is not granted for all kids, so, my focus is to introduce programming concepts gradually in parallel with math concepts, so all students can find it useful even they don't want to be programmers or think that AI will write programs for them.
Also, there are many games written with Small Basic, and with little modifications they can run in sVB. On the other hand, sVB focusses on Windows forms, which has a lot of info to be introduced to kids, and this can't happen through game samples. In fact, even simple games need to introduce arrays, if conditions and loops, which are not suitable for most kids under 10 years, because their logical thinking depends on age and math knowledge. This is why I am starting the book series with visual programmibg and basic OOP concepts with minimal posdiple code lines. But be sure that the content you want will be introduced in the series (I am planing for 12 to 20 small books, 100 to 150 pages each, more than half of the size is pictures, sammeries, questions and excersies).
Believe it or not,  In Egypt, Scratch is introduced in schools in the 7th grade! I am trying to make something different here, to buildup a programmer from ground up, while helping students to be better in logic, math and science. This is why I am against to add the LitDev library to sVB, because playing with advanced tools easily will make them learn nothing!
In the near future, all programmers will need a deep knowledge in math and electronics to work in the AI fields.
RAD programming, including desktop, web and mobile aps, is expected to be fully done by AI apps and no developers will be needed in these areas. Even complex games can be easily auto generated with AI. So, if there will be any programmers left, they will be the ones that can outsmart the AI
So, sVB is not about to reinventing SB, VB, Python or Scratch. It is meant to be eaier than all of these languages to gradually introduce a real programming and math concepts to kids, to be better students and potential future programmers.
Thanks.


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: wc-yaa ***@***.***>
Sent: Friday, January 26, 2024 6:16:16 AM
To: VBAndCs/sVB-Small-Visual-Basic ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [VBAndCs/sVB-Small-Visual-Basic] Two amateur questions. Samples and book. (Discussion #52)


Hello, I have two questions.

There are many nice samples in the package. Really good. Some of them are games(ball game, cars, etc.) If several famous classic games are also placed in the package, children may easily be motivated to try to learn this BASIC programming language. For example, Tic-tac-toe, Tetris, Snake, Asteroids, Pong, etc. Let children know the versatility of this BASIC language. In the past, they probably only thought that only GameMaker, Scratch or Godot could be used to make games.

Another question is about book for children. There is a nice Python book for children. The book was published many years ago and is quite well-known. The book teaches readers how to make about 10 mini-games. In the process of learning how to make each game, the readers can also learn several programming concepts. The book can be bought in online shops but the HTML version can be read here:
https://inventwithpython.com/invent4thed/

This arrangement is nice. Children can read the book freely. This is also an effective promotion for the book and Python. Right?
If the author of Small Visual Basic plans to write a new book, this type of book may be a good option.

The author of Small Visual Basic may be too busy in developing Small Visual Basic. Thus I really hope that some experienced Small Visual Basic users will also consider the above idea. It is a meaningful concept.

(Actually, the writer wrote another book several years later: https://inventwithpython.com/pygame/)

**Comment by wc-yaa** [comment]
Hello! Yur post shows that you have a complete plan. Good job!

>Thanks for your passion for sVB. 

Only some unbiased opinions from an amateur. 

>so, my focus is to introduce programming concepts gradually in parallel with math concepts, so all students can find it useful...."

I totally agree.

>But be sure that the content you want will be introduced in the series (I am planing for 12 to 20 small books,"
 
Your plan is really big!

>So, sVB is not about to reinventing SB, VB, Python or Scratch."

I understand now. Yeah.

I also want to talk about the user base. If there is a sVB book(similar to the two HTML Python books) placed on your website, some young visitors(clever kids or teenagers(not your students)) will be able to read the book to learn by themselves. These kids and teenagers may be much more than your students in schools. As a result, the user base of sVB will be enlarged significantly and more quickly. In fact, I discovered that so many BASIC languages only had tiny user bases.

Anyway, good luck!!!!

**Comment by VBAndCs** [comment]
I have introduced sVB while I am creating it step by a step in a video series but in Arabic, then re-introduces the series for kids in 1 minutes videos (also in Arabic), before writing the full reference book and now creating the kids and beginners series. Obviously, I can't do more, due to my limited time, so, I asked content makers to create videos about sVB in their native languages. This will be faster that books to reach beginners.

I like the idea of the two books and I may write one of them some day, but I am still evolving sVB. In the last updates, sVB form designer had a properties window, and in the upcoming sVB 2.9, it will has a menu designer. By thee way, you can currently create menus by code (see the sVB notepad project in the samples folder).

Thanks.


________________________________

**Reply by wc-yaa** [reply]
(sorry for my late reply)

Thanks! Thanks for considering my idea again. I hope that every reader of the book will become happy user of Small Visual Basic. Sure. Good luck.

**Comment by VBAndCs** [comment]
I just published the first book of the "Small Visual Basic Kid Programmer" series on Amazon: https://www.amazon.com/dp/B0CVK1BMC9
________________________________
